sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosDataWriteFactory.scala

```diff
@@ -25,27 +27,81 @@ class CosmosDataWriteFactory(userConfig: Map[String, String],
 
     val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(userConfig)
     val cosmosTargetContainerConfig = CosmosContainerConfig.parseCosmosContainerConfig(userConfig)
+    val cosmosWriteConfig = CosmosWriteConfig.parseWriteConfig(userConfig)
 
     // TODO moderakh: this needs to be shared to avoid creating multiple clients
     val builder = new CosmosClientBuilder()
       .key(cosmosAccountConfig.key)
       .endpoint(cosmosAccountConfig.endpoint)
-      .consistencyLevel(ConsistencyLevel.EVENTUAL);
+      .consistencyLevel(ConsistencyLevel.EVENTUAL)
 
     SparkBridgeInternal.setMetadataCacheSnapshot(builder, cosmosClientStateHandle.value)
     val client = builder.buildAsyncClient();
 
+    val container = client.getDatabase(cosmosTargetContainerConfig.database)
+      .getContainer(cosmosTargetContainerConfig.container)
+
+    val containerDefinition = container.read().block().getProperties
+    val partitionKeyDefinition = containerDefinition.getPartitionKeyDefinition
```

Comment:
What happens if the collection is deleted and recreated with a different PK? I think the SDK handles that behavior, but this value would be already cached, right?

Reply:
this will get initialized per dataframe instantiation. collection cannot be deleted in between while the dataframe is in use.